### ISSUE TITLE
Errors during golangci operation

### ISSUE THREAD

**Issue by ids79** [golangci/golangci-lint#3408]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc.).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

I am using the last version golangco 1.50.1 on local maschine (windows). 
When I try to perform linter, I get following errors: 
```
level=warning msg="[runner] Can't run linter goanalysis_metalinter: gci: 4:12: expected ';', found \"testing\" (and 3 more errors)"
level=error msg="Running error: 1 error occurred:\n\t* can't run linter goanalysis_metalinter: gci: 4:12: expected ';', found \"testing\" (and 3 more errors)\n\n"
```

What can they be related to?
How can I fix them?

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
# Paste output here
```
1.50.1
</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
# paste output here
```
```yml
run:
  tests: true
  build-tags:
    - bench
    - !bench

linters-settings:
  funlen:
    lines: 150
    statements: 80

issues:
  exclude-rules:
    - path: _test\.go
      linters:
        - errcheck
        - dupl
        - gocyclo
        - gosec

linters:
  disable-all: true
  enable:
    - asciicheck
    - bodyclose
    - depguard
    - dogsled
    - dupl
    - durationcheck
    - errorlint
    - exhaustive
    - exportloopref
    - funlen
    - gci
    - gocognit
    - goconst
    - gocritic
    - gocyclo
    - godot
    - gofmt
    - gofumpt
    - goheader
    - goprintffuncname
    - gosec
    - gosimple
    - govet
    - importas
    - ineffassign
    - lll
    - makezero
    - misspell
    - nestif
    - nilerr
    - noctx
    - nolintlint
    - prealloc
    - predeclared
    - revive
    - rowserrcheck
    - sqlclosecheck
    - staticcheck
    - stylecheck
    - tagliatelle
    - thelper
    - typecheck
    - unconvert
    - unparam
    - unused
    - wastedassign
    - whitespace
```

</details>

### Go environment

<details>

```console
$ go version && go env
# paste output here
```
windows
</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v

level=info msg="[config_reader] Config search paths: [./ C:\\Users\\divanov\\Desktop\\GO\\otus_go_homework C:\\Users\\divanov\\Desktop\\GO C:\\Users\\divanov\\Desktop C:\\Users\\divanov C:\\Users C:\\ C:\\Users\\divanov\\go]"
level=info msg="[config_reader] Used config file .golangci.yml"
level=info msg="[lintersdb] Active 47 linters: [asciicheck bodyclose depguard dogsled dupl durationcheck errorlint exhaustive exportloopref funlen gci gocognit goconst gocritic gocyclo godot gofmt gofumpt goheader goprintffuncname gosec gosimple govet importas ineffassign lll makezero misspell nestif nilerr noctx nolintlint prealloc predeclared revive rowserrcheck sqlclosecheck staticcheck stylecheck tagliatelle thelper typecheck unconvert unparam unused wastedassign whitespace]"
level=info msg="[loader] Using build tags: [bench ]"
level=info msg="[loader] Go packages loading at mode 575 (compiled_files|deps|files|imports|exports_file|name|types_sizes) took 243.6534ms"
level=info msg="[runner/filename_unadjuster] Pre-built 0 adjustments in 0s"
level=info msg="[linters_context] importas settings found, but no aliases listed. List aliases under alias: key."
level=error msg="[linters_context] typechecking error: pattern ./...: directory prefix . does not contain modules listed in go.work or their selected dependencies"
level=info msg="[linters_context/goanalysis] analyzers took 6.9762ms with top 10 stages: buildir: 4.9827ms, fact_deprecated: 1.9935ms, cgocall: 0s, ST1020: 0s, gosec: 0s, SA5002: 0s, SA6001: 0s, exportloopref: 0s, printf: 0s, SA1029: 0s"
level=info msg="[runner] processing took 0s with stages: exclude-rules: 0s, nolint: 0s, max_per_file_from_linter: 0s, severity-rules: 0s, path_prefixer: 0s, sort_results: 0s, filename_unadjuster: 0s, path_prettifier: 0s, skip_dirs: 0s, diff: 0s, max_same_issues: 0s, max_from_linter: 0s, autogenerated_exclude: 0s, exclude: 0s, uniq_by_line: 0s, source_code: 0s, path_shortener: 0s, cgo: 0s, skip_files: 0s, identifier_marker: 0s"
level=info msg="[runner] linters took 1.0423697s with stages: goanalysis_metalinter: 1.041954s"
level=info msg="File cache stats: 0 entries of total size 0B"
level=info msg="Memory: 15 samples, avg is 46.3MB, max is 56.9MB"
level=info msg="Execution took 1.3529325s"
```

</details>

### Code example or link to a public repository

<details>

```go
// add your code here
```
</details>

**Comment by ldez** [comment]
@ids79 @matjam have you tried with previous versions? (v1.50.0, v1.49.0, ...)

**Comment by ids79** [comment]
Yes I have tried since 1.48.0 version. The Result the same.

**Comment by ldez** [comment]
So it seems to be a missing piece inside your OS.

Can you try the following commands and give me the output:
- `diff --help`
- `diff.exe --help`

**Comment by ids79** [comment]
PS C:\Users\divanov\Desktop\GO\otus\struct> diff --help

Командлет Compare-Object в конвейере команд в позиции 1
Укажите значения для следующих параметров:
DifferenceObject[0]:

**Comment by alexandear** [comment]
This issue appears because Windows doesn't have [`diff`](https://man7.org/linux/man-pages/man1/diff.1.html) tool. `diff` is needed to compute the difference used in `gofmt` internal [`diff` function](https://github.com/golangci/gofmt/blob/f2edd75033f2ca22191625f8286bf50bca0d45ae/goimports/goimports.go#L48).

A workaround for the issue is to append the directory with the program `diff.exe` to the `Path` environment variable. See https://stackoverflow.com/a/56206563. Assuming, you have `git` which contains `diff.exe` in `C:\Program Files\Git\usr\bin`. PowerShell command to modify `Path` user variable is the following:
```PowerShell
[Environment]::SetEnvironmentVariable("Path", $env:Path + ";C:\Program Files\Git\usr\bin", "User")
```

Some may disagree that Windows has PowerShell with working `diff`. But it's an alias to PowerShell's [`Compare-Object`](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/compare-object?view=powershell-7.3). `Compare-Object` produces output in a different format than `diff` in Unix-like systems.

**Comment by enumag** [comment]
My colleague ran into this issue as well. He's working on Windows but the odd thing is that he's running go and golangci-lint inside a docker container.

Admittedly `diff` is missing in the docker container but if it was required then it should fail for me as well right? I'm running the linter inside docker as well, just my host system is ubuntu.

Why does linter behave differently when running inside docker?

**Comment by enumag** [comment]
Ah okay it seems it's actually not a problem of the golangci-lint itself but rather `gci` in particular. So I disabled it in favor of `goimports`.

**Comment by Antonboom** [comment]
Related to https://github.com/golangci/golangci-lint/issues/362

**Comment by trollixx** [comment]
Combination of the following pull requests addresses this issue on Windows when `diff.exe` is available:

- https://github.com/golangci/golangci-lint/pull/4113
- https://github.com/golangci/gofmt/pull/5
- https://github.com/golangci/gofmt/pull/6
